ASP-NET/ASP-NET/Controllers/Auth.cs

```diff
@@ -0,0 +1,113 @@
+﻿using ASP_NET.Context;
+using ASP_NET.Dto;
+using ASP_NET.Entities;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.IdentityModel.Tokens;
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Text;
+
+namespace ASP_NET.Controllers
+{
+    [Route("api/auth")]
+    [ApiController]
+    public class AuthController : ControllerBase
+    {
+
+        private readonly ApplicationDbContext _context;
+        private readonly UserManager<User> _userManager;
+        private readonly IConfiguration _configuration;
+
+        public AuthController(ApplicationDbContext context, UserManager<User> userManager, IConfiguration configuration)
+        {
+            _context = context;
+            _userManager = userManager;
+            _configuration = configuration;
+        }
+
+
+        [HttpPost("register")]
+        public async Task<ActionResult<User>> Register(RegisterDto request)
+        {
+            var passwordHash = BCrypt.Net.BCrypt.HashPassword(request.Password);
+
+
+            User user = new User { Email = request.Email, PasswordHash = passwordHash, UserName = request.Username };
+
+            var result = await this._userManager.CreateAsync(user);
+
+            return Ok(user);
+        }
+
+
+        [HttpPost("login")]
+        public async Task<ActionResult<ResultResponseDto>> Login(LoginDto request)
+        {
+            Console.WriteLine(request.Email);
+
+            var user = await this._userManager.FindByEmailAsync(request.Email);
```

Comment:
Please follow N-layer architecture principles and move all the business logic to application layer(create services) in all the controllers